### PR TITLE
Create user interface for managing bookings

### LINKED ISSUE #19: FE: Manage Bookings UI
This view is for pet sitters who have requests coming in and have the ability to manage their bookings. 

Follow mockup. Refer to "Manage bookings" in the spec document

The mockup should include: 
-Calendar view
\- active upcoming bookings

### PR DESCRIPTION
**Pull request by michael-levitski**
Closes  #19

### What this PR does (required):
- list PR functionality here

### Screenshot
Screenshot 2022-01-17 190630

### Any information needed to test this feature (required):
- uses `calendar` npm package
- visit url path `/bookings`

### CODE REVIEW

File: client/src/components/Calendar/CalendarBody.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  const calendar = new Calendar();
+  const year = selectedMonth.getFullYear();
+  const month = selectedMonth.getMonth();
+
+  // the following 3 arrays are 2d arrays
+  // outer arrays represent weeks in the month
+  // all inner arrays will have length of 7, grouping dates into weeks
+
+  // daysOfMonth numbers will either represent the date, or 0
+  //if the date falls outside of the current month
```
Comment (sundayezeilo) [inline client/src/components/Calendar/CalendarBody.tsx]: Please remove these comments

File: client/src/components/Calendar/CalendarDay.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    }
+    return style;
+  },
+}));
+
+const CalendarDay = ({ calendarDate }: PropTypes) => {
+  const classes = useStyles(calendarDate);
+  const numDate = calendarDate.date.getDate();
+  return (
+    <div style={{ width: '50px', textAlign: 'center', margin: '5px 0px' }}>
```
Comment (sundayezeilo) [inline client/src/components/Calendar/CalendarDay.tsx]: Please use the MUI `Box` components for `div`

File: client/src/pages/Bookings/Bookings.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    },
+    start: new Date('2022-01-16T15:00'),
+    end: new Date('2022-01-16T22:00'),
+    status: 'accepted',
+  },
+];
+
+const combinedBookings = [...inMemoryCurrentBookings, ...inMemoryPastBookings, inMemoryNextBooking];
+const filteredBookings = combinedBookings.filter((b) => b.status !== 'declined');
+console.log(filteredBookings.map((b) => b.start));
```
Comment (sundayezeilo) [inline client/src/pages/Bookings/Bookings.tsx]: Please watch for unnecessary logs to the console. Going forward, we should be cleaning up our logs done during development. Thanks!

File: client/src/pages/Bookings/Bookings.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+const highlightedDates = filteredBookings.map((b) => b.start.toString().slice(0, 10));
+console.log(highlightedDates);
```
Comment (sundayezeilo) [inline client/src/pages/Bookings/Bookings.tsx]: The same comment for console.log

**Review by sundayezeilo — CHANGES_REQUESTED** [review]
Good work on this one. But a few comments to have a look at.

File: client/src/components/BookingList/Booking.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import SettingsIcon from '@mui/icons-material/Settings';
+import format from 'date-fns/format';
+
+interface PropTypes {
+  booking: IBooking;
+}
+
+const Booking = ({ booking }: PropTypes) => {
+  return (
+    <Card variant="outlined" sx={{ padding: '0px 10px 10px 10px', margin: '10px 0px' }}>
```
Comment (rajivtitus) [inline client/src/components/BookingList/Booking.tsx]: 👍🏽

File: client/src/components/BookingList/Booking.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import { Booking as IBooking } from '../../interface/Booking';
+import { Avatar, Card, Grid, IconButton, Typography, Box } from '@mui/material';
+import SettingsIcon from '@mui/icons-material/Settings';
+import format from 'date-fns/format';
+
+interface PropTypes {
+  booking: IBooking;
```
Comment (rajivtitus) [inline client/src/components/BookingList/Booking.tsx]:
```suggestion
  booking: Booking;
```
We don't need to prefix our interfaces with 'I' in JS. Ditto where applicable!

File: client/src/components/BookingList/Booking.tsx (hunk 5 above)
Comment (rajivtitus) [inline client/src/components/BookingList/Booking.tsx]:
```suggestion
    <Card variant="outlined" sx={{ padding: '0px 10px 10px 10px', margin: '10px 0px' }}>
      <CardContent>
```
Nit: We can wrap the contents with the `CardContent` component to stay consistent with MUI semantics 😄

File: client/src/components/BookingList/BookingList.tsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import { Typography } from '@mui/material';
+import { Booking as IBooking } from '../../interface/Booking';
+import Booking from './Booking';
+
+export interface PropTypes {
+  bookings: IBooking[];
```
Comment (rajivtitus) [inline client/src/components/BookingList/BookingList.tsx]:
```suggestion
  bookings: Booking[];
```

File: client/src/components/BookingList/BookingList.tsx [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+const BookingList = ({ bookings, type }: PropTypes) => {
+  return (
+    <>
+      <Typography sx={{ marginTop: '10px', fontVariant: 'small-caps', fontWeight: 'bold' }}>
+        {type} bookings:
+      </Typography>
+      {bookings.map((b) => (
+        <Booking key={b.start.getTime()} booking={b} />
```
Comment (rajivtitus) [inline client/src/components/BookingList/BookingList.tsx]: Let's make sure we use unique indexes as keys in React. If two pending bookings have the same start time we will no longer have unique keys for our component

Importance of unique indexes in React: https://betterprogramming.pub/why-react-keys-matter-an-introduction-136b7447cefc

File: client/src/components/Calendar/CalendarDay.tsx [hunk 9]
```diff
@@ -0,0 +1,36 @@
+import { Badge } from '@mui/material';
+import { makeStyles } from '@mui/styles';
+import { CalendarDate } from '../../interface/Calendar';
+
+interface PropTypes {
+  calendarDate: CalendarDate;
+}
+
+const useStyles = makeStyles((theme: any) => ({
```
Comment (rajivtitus) [inline client/src/components/Calendar/CalendarDay.tsx]: Please avoid using `any` as a type

File: client/src/components/Calendar/CalendarDay.tsx [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+        color: 'white',
+        backgroundColor: theme.palette.primary.main,
+      };
+    }
+    return style;
+  },
+}));
+
+const CalendarDay = ({ calendarDate }: PropTypes) => {
+  const classes = useStyles(calendarDate);
```
Comment (rajivtitus) [inline client/src/components/Calendar/CalendarDay.tsx:28]: 👍🏽

File: client/src/components/NextBookings/NextBooking.tsx [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+import { Booking } from '../../interface/Booking';
+
+interface PropTypes {
+  booking: Booking;
+}
+
+const NextBooking = ({ booking }: PropTypes) => {
+  return (
+    <Card sx={{ padding: '10px 0px 30px 20px' }}>
+      <Grid container direction="row" justifyContent="space-between" alignItems="flex-end">
```
Comment (rajivtitus) [inline client/src/components/NextBookings/NextBooking.tsx]: Same as comment above about the usage of `CardContent`

File: client/src/pages/Bookings/Bookings.tsx [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+  start: new Date('2022-01-20T10:00'),
+  end: new Date('2022-01-20T12:00'),
+  status: 'accepted',
+  user: {
+    name: 'Norma Byes',
+    email: 'norma.byes@example.com',
+  },
+};
+
+const inMemoryCurrentBookings: Booking[] = [
```
Comment (rajivtitus) [inline client/src/pages/Bookings/Bookings.tsx]: 👍🏽

File: client/src/pages/Bookings/Bookings.tsx [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+console.log(filteredBookings.map((b) => b.start));
+const highlightedDates = filteredBookings.map((b) => b.start.toString().slice(0, 10));
+console.log(highlightedDates);
+
+const Bookings = () => {
+  const { loggedInUser } = useAuth();
+  const { initSocket } = useSocket();
+  const history = useHistory();
+
+  useEffect(() => {
```
Comment (rajivtitus) [inline client/src/pages/Bookings/Bookings.tsx:39]: Please remove this `useEffect`, we don't need to initialize the socket connection within this component

File: client/src/pages/Bookings/Bookings.tsx [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+      <Grid container justifyContent="space-evenly">
+        <Grid item xs={4}>
+          <NextBooking booking={inMemoryNextBooking} />
+          <Box sx={{ height: '20px' }} />
+          <Card
+            sx={{
+              padding: '0px 10px 10px 10px',
+              height: '300px',
+              overflow: 'auto',
+              '::-webkit-scrollbar': {
```
Comment (rajivtitus) [inline client/src/pages/Bookings/Bookings.tsx]: Suggestion: I think we can move some of these additional styles into a CSS class for increased readability within this component

**Review by rajivtitus — COMMENTED** [review]
Very good work with this Michael! Just some minor comments for you to address, thanks!

File: client/src/pages/Bookings/Bookings.tsx [hunk 15]
```diff
@@ -54,18 +57,26 @@ const inMemoryPastBookings: Booking[] = [
...
+    height: '300px',
+    overflow: 'auto',
+    '::-webkit-scrollbar': {
+      width: 6,
+      bgcolor: 'rgba(0, 0, 0, 0.26)',
+      borderRadius: 2,
+    },
+    '::-webkit-scrollbar-thumb': { bgcolor: 'rgba(0, 0, 0, 0.21)', borderRadius: 2 },
+  },
+});
```
Comment (michael-levitski) [inline client/src/pages/Bookings/Bookings.tsx]: This change is outdated due to a future commit

File: client/src/pages/Bookings/Bookings.tsx [hunk 16]
```diff
@@ -79,19 +90,7 @@ const Bookings = () => {
...
-              overflow: 'auto',
-              '::-webkit-scrollbar': {
-                width: 6,
-                bgcolor: 'rgba(0, 0, 0, 0.26)',
-                borderRadius: 2,
-              },
-              '::-webkit-scrollbar-thumb': { bgcolor: 'rgba(0, 0, 0, 0.21)', borderRadius: 2 },
-            }}
-          >
+          <Card className={classes.scrollableCard}>
```
Comment (michael-levitski) [inline client/src/pages/Bookings/Bookings.tsx]: This change is outdated due to a future commit

**Review by rajivtitus — APPROVED** [review]
Excellent work with those changes 💯

**Pull request merged**